### PR TITLE
reef: common/admin_socket: add a command to raise a signal

### PR DESCRIPTION
**Pull request by leonid-s-usov**
Backport ticket: https://tracker.ceph.com/issues/63363
Original ticket: https://tracker.ceph.com/issues/62882
Original PR: https://github.com/ceph/ceph/pull/53689

---

The new command "raise <signal> [--after X]" accepts signals in the forms: '9', '-9', 'kill', '-KILL'
When --after is specified, the program will fork to wait for the timeout

The forked instance will bail out if it detects that the parent PID has changed which would indicate that the original parent has terminated.

Forking an instance allows to schedule delivery of signals even if the original process is suspended, e.g.:
```
  ceph tell mds.a raise CONT --after 10
  ceph tell mds.a raise STOP
```

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by ronen-fr — APPROVED** [review]

**Comment by ronen-fr** [comment]
The compilation issue: a missing fmt include?

**Comment by leonid-s-usov** [comment]
There are more errors than just that. I didn't expect it.. apparently, the compilation settings are different across release branches.

File: src/common/admin_socket.cc [hunk 1]
```diff
@@ -693,6 +695,298 @@ class GetdescsHook : public AdminSocketHook {
...
+      }
+      f->close_section();
+    } else {
+      try {
+        result = std::stoi(sigdesc);
+        if (result < 1 || result > 64) {
+          errss << "signal number should be an integer in the range [1..64]" << std::endl;
+          return -EINVAL;
+        }
+      } catch (std::invalid_argument) {
```
Comment (cbodley) [inline src/common/admin_socket.cc:870]: this introduces a compiler warning, see https://tracker.ceph.com/issues/63455 https://github.com/ceph/ceph/pull/54365

Comment (leonid-s-usov) [reply src/common/admin_socket.cc:870]: ack, thanks!

**Comment by leonid-s-usov** [comment]
jenkins test windows

**Comment by leonid-s-usov** [comment]
jenkins test windows

**Comment by leonid-s-usov** [comment]
jenkins test api

**Comment by vshankar** [comment]
@leonid-s-usov just confirming - this backport has fixes from https://github.com/ceph/ceph/pull/54451, yes?

**Comment by leonid-s-usov** [comment]
> @leonid-s-usov just confirming - this backport has fixes from #54451, yes?

That is correct. It's actually the other way around: #54451 has been created to apply the changes that were implemented in these backports

**Comment by Matan-B** [comment]
RADOS approved  https://tracker.ceph.com/issues/65574

**Review by anthonyeleven — APPROVED** [review]

**Pull request merged**